.github/workflows/ci.yml

```diff
@@ -73,15 +73,19 @@ jobs:
             os: ubuntu-20.04  # CPython 3.9.2 is not available for ubuntu-22.04.
             experimental: false
             nox-session: test-3.9
-          - python-version: "3.11"
+          - python-version: "3.12"
             os: ubuntu-latest
-            nox-session: emscripten
+            nox-session: emscripten(firefox)
```

Comment:
We have to change the session name in the steps installing browsers
https://github.com/urllib3/urllib3/blob/756f8b8fa85ad1807047c15bea766e1a8f1644ab/.github/workflows/ci.yml#L104-L109

It's interesting that the tests pass without installing browsers at the moment
![](https://github.com/user-attachments/assets/fd80bded-2cab-4f12-800a-b5ead647428d)

Reply:
Good catch!

@joemarshall Any idea if this is actually needed or not?

Reply:
Looking at https://github.com/urllib3/urllib3/pull/3427/files, it looks like we have an use case for specifying a version, but then there's a lot of ln and rm happening to actually use it?

That said, to avoid blocking this pull request, I fixed the issue. And then ran into a new Selenium warning introduced today, that I reported at https://github.com/SeleniumHQ/selenium/issues/14622 and added to our warning filters list.